packages/ui/app/src/api-page/endpoints/EndpointContent.tsx

```diff
@@ -84,6 +84,7 @@ export const EndpointContent = React.memo<EndpointContent.Props>(function Endpoi
         },
         [titleHeight]
     );
+    const [selectedErrorIndex, setSelectedErrorIndex] = useState<number | null>(null);
```

Comment:
nbd but everywhere else in the UI we never use null / always use undefined

Reply:
also, jw why is this state hoisted here in EndpointContent and not just in EndpointErrorsSection?

Reply:
re `null` good to know, will switch to `undefined` for consistency

Reply:
re hoisted state, I default to keeping components stateless. As far as I know, it's generally recommended to have as few stateful components as possible as it's easier to reuse, test and debug them that way. Particularly true if we were using Storybook to build these components